### PR TITLE
Make plugin hook more explicit

### PR DESCRIPTION
**Pull request by oyvindeide**
@dafeda here is an example rewrite which is more explicit. Though this is logging, so will always be possible to things globally.

### CODE REVIEW

File: ert_shared/plugins/hook_specifications/logging.py [hunk 1]
```diff
@@ -2,11 +2,9 @@
...
-def add_log_handle(logging):
+def add_log_handle_to_root():
     """
-    Modify ert's logging instance, for example adding handlers and changing formatting.
+    Adding handlers to the root logger in ert.
 
-    :param logging: A handle to the ert logging instance.
-    :type logging: :module:`logging`
-    :return: None
+    :return: logging.Handler
```
Comment (dafeda) [inline ert_shared/plugins/hook_specifications/logging.py:9]: Should this be a list of logging.Handler(s) instead?

File: ert_shared/plugins/hook_specifications/logging.py [hunk 2]
```diff
@@ -2,11 +2,9 @@
 
 
 @hook_specification
-def add_log_handle(logging):
+def add_log_handle_to_root():
     """
-    Modify ert's logging instance, for example adding handlers and changing formatting.
+    Adding handlers to the root logger in ert.
```
Comment (dafeda) [inline ert_shared/plugins/hook_specifications/logging.py:7]: Propsed change to docstring to hopefully make it more clear to those implementing hooks what we expect.

Return a list of one or multiple log handlers that you want added to ERT's root-logger.

File: tests/ert_tests/all/plugins/test_plugin_manager.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def test_workflows_merge_duplicate(caplog):
 def test_add_logging_handle(tmpdir):
     with tmpdir.as_cwd():
         pm = ErtPluginManager(plugins=[dummy_plugins])
-        pm.add_logging_handles(logging)
+        pm.add_logging_handle_to_root(logging.getLogger())
         logging.critical("I should write this to spam.log")
         with open("spam.log") as fin:
```
Comment (dafeda) [inline tests/ert_tests/all/plugins/test_plugin_manager.py:136]: Propose explicitly specifying encoding to utf-8.

File: ert_shared/plugins/hook_specifications/logging.py (hunk 1 above)
Comment (oyvindeide) [reply ert_shared/plugins/hook_specifications/logging.py:9]: No, pluggy will automatically add return values to a list as there could be several plugins serving up different handlers.

File: ert_shared/plugins/hook_specifications/logging.py (hunk 2 above)
Comment (oyvindeide) [reply ert_shared/plugins/hook_specifications/logging.py:7]: Agree that we can improve the doc string, but we can not return a list, you would have to add a single handler per plugin. I think that is ok, but that is a change we make intentionally ;)

Comment (dafeda) [reply ert_shared/plugins/hook_specifications/logging.py:7]: Ah I see, so something like "Return a log handler that you want added to ERT's root-logger"?

**Review by dafeda — APPROVED** [review]
LGTM :+1:

**Pull request merged**